pkg/gatewayconfigurationserver/grpc.go

```diff
@@ -0,0 +1,127 @@
+// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gatewayconfigurationserver
+
+import (
+	"context"
+	"encoding/json"
+
+	types "github.com/gogo/protobuf/types"
+	"go.thethings.network/lorawan-stack/v3/pkg/auth/rights"
+	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+	"go.thethings.network/lorawan-stack/v3/pkg/frequencyplans"
+	"go.thethings.network/lorawan-stack/v3/pkg/pfconfig/cpf"
+	"go.thethings.network/lorawan-stack/v3/pkg/pfconfig/semtechudp"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+)
+
+var errInvalidConfigurationFormat = errors.DefineInvalidArgument("invalid_configuration_format", "configuration format `{format}` is not supported")
+var errInvalidConfigurationType = errors.DefineInvalidArgument("invalid_configuration_type", "configuration type `{type}` is not supported")
+var errInvalidConfigurationFilename = errors.DefineInvalidArgument("invalid_configuration_filename", "configuration filename `{filename}` is not supported")
```

Comment:
Nit. Since the error type is already `InvalidArgument`, I'd recommend replacing the word `Invalid` with `Unsupported` or `NotSupported` in the variable and the error name.